src/catalog/catalog_entry/function_catalog_entry.cpp

```diff
@@ -0,0 +1,24 @@
+#include "catalog/catalog_entry/function_catalog_entry.h"
+
+namespace kuzu {
+namespace catalog {
+
+FunctionCatalogEntry::FunctionCatalogEntry(
+    CatalogEntryType entryType, std::string name, function::function_set functionSet)
+    : CatalogEntry{entryType, std::move(name)}, functionSet{std::move(functionSet)} {}
+
+std::unique_ptr<CatalogEntry> FunctionCatalogEntry::copy() const {
+    return std::make_unique<FunctionCatalogEntry>(
+        getType(), getName(), copyFunctionSet(functionSet));
+}
+
+function::function_set FunctionCatalogEntry::copyFunctionSet(const function::function_set& set) {
```

Comment:
Can we do something similar as in copy_constructor.h to template copy vectors of unique_ptr?

```
template<typename T>
static std::vector<T> copyVector(const std::vector<T>& objects) {
    std::vector<T> result;
    result.reserve(objects.size());
    for (auto& object : objects) {
        result.push_back(object.copy());
    }
    return result;
}
```